src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheck.java

```diff
@@ -386,6 +388,7 @@ public int[] getDefaultTokens() {
             TokenTypes.BNOT,
             TokenTypes.POST_INC,
             TokenTypes.POST_DEC,
+            TokenTypes.LPAREN,
```

Comment:
IMO, adding `LPAREN` as an acceptable token to this check feels odd.  See all the tokens in this array, you will notice that they are operators. So, I expected the fix to be by adding `TokenTypes.QUESTION`, Have you tried this way?

Reply:
Honestly I tried both options yes but thought that there would be a problem with inner conditional expressions like `(a > b) ? //doSth1 : (c > d) ? //doSth2 : //doSth3`
But after visualizing the AST again turns out `Question` would still be a parent to an `LPAREN` even if it's an inner expression.
So I think it wouldnt matter both ways but you are right it's a bit odd to have `LPAREN` i should probably switch it to have `QUESTION` and check over its children to check if exists any 'LPAREN` then i'd log it.
 WDYT ?

Reply:
`y == 3 ? (3) : (4) `  we have to think about paren in this example also. This will most likely get covered under `QUESTION` also

```
        |       |           `--QUESTION -> ? [3:25]
        |       |               |--LPAREN -> ( [3:16]
        |       |               |--EQUAL -> == [3:19]
        |       |               |   |--IDENT -> y [3:17]
        |       |               |   `--NUM_INT -> 3 [3:22]
        |       |               |--RPAREN -> ) [3:23]
        |       |               |--LPAREN -> ( [3:27]
        |       |               |--NUM_INT -> 3 [3:28]
        |       |               |--RPAREN -> ) [3:29]
        |       |               |--COLON -> : [3:31]
        |       |               |--LPAREN -> ( [3:33]
        |       |               |--NUM_INT -> 4 [3:34]
        |       |               `--RPAREN -> ) [3:35]
```

Reply:
> !(a>b) ? 3 : 4 
> In this example, this is not unnecessary parens.

I think this is not a problem because LPAREN will not be a direct child to QUESTION so it will not be caught as unnecessary parens

```
        |           `--QUESTION -> ? [3:17]
        |       |               |--LNOT -> ! [3:10]
        |       |               |   |--LPAREN -> ( [3:11]
        |       |               |   |--GT -> > [3:13]
        |       |               |   |   |--IDENT -> a [3:12]
        |       |               |   |   `--IDENT -> b [3:14]
        |       |               |   `--RPAREN -> ) [3:15]
        |       |               |--NUM_INT -> 3 [3:19]
        |       |               |--COLON -> : [3:21]
        |       |               `--NUM_INT -> 4 [3:23]
```
 
 > This will most likely get covered under QUESTION also
 
 
Yup it will not be a problem any direct LPAREN is always an unnecessary one **even if there is a ``!`` it will be the one directly below ``QUESTION``**

So do you think this is the appropriate way ? To check if there is a direct LPAREN child covererd below QUESTION ? IMO and as far as I have tried visualizing all the possible cases it's always the case , QUESTION -> LPAREN (direct child)

Reply:
IMO, the best way to be sure is to code, create an input file with many edge cases, and run tests. This way everything will be more concrete and we can spot any gaps. we can always improve based on what we find.

Reply:
done 
Added QUESTION instead and covered its cases and also added a new input file , although a lot of the edge cases are covered in many files as well